### DISCUSSION TITLE
Codespace stopping unexpectedly

### DISCUSSION THREAD

**Discussion by MFAshby** [community/community#5398] [Codespaces]
Hi, 

I have a 32 core codespace [1] and I'm occasionally running some quite intensive applications (a microservices stack of about ~10 java web apps). Several times today, my codespace has suddenly stopped; I get the message 'Codespace is stopping' and I have to restart it. It's quite annoying as all my docker containers also stop and I have to reboot all the apps to begin work again.

[1] https://mfashby-patientsknowbest-devenv-64x74rr2x55q.github.dev/

**Comment by tanmayeekamath** [comment]
Hey @MFAshby, the current default auto-stop timeout for Codespaces is 30 mins. To help with saving costs we automatically stop a running codespace if we don't detect any activity for a period of 30 mins (more on that here: https://docs.github.com/en/codespaces/getting-started/deep-dive#saving-files-in-your-codespace)
Can you confirm if your Codespace stop behavior is inline with above? 

Having said that, we have a workitem on our roadmap that will enable configuring your own default timeouts at an org level to be inline with your scenarios that should be available soon.
